Modules/Synchronization/src/Synchronization.ConsumerApi/Controllers/SyncRunsController.cs

```diff
@@ -26,20 +29,27 @@ namespace Backbone.Modules.Synchronization.ConsumerApi.Controllers;
 public class SyncRunsController : ApiControllerBase
 {
     private readonly ApplicationOptions _options;
+    private readonly IUserContext _userContext;
+    private readonly IIdentitiesRepository _identitiesRepository;
 
-    public SyncRunsController(IMediator mediator, IOptions<ApplicationOptions> options) : base(mediator)
+    public SyncRunsController(IMediator mediator, IOptions<ApplicationOptions> options, IUserContext userContext, IIdentitiesRepository identitiesRepository) : base(mediator)
     {
         _options = options.Value;
+        _userContext = userContext;
+        _identitiesRepository = identitiesRepository;
     }
 
-
     [HttpPost]
     [ProducesResponseType(typeof(HttpResponseEnvelopeResult<StartSyncRunResponse>), StatusCodes.Status200OK)]
     [ProducesError(StatusCodes.Status400BadRequest)]
     public async Task<IActionResult> StartSyncRun(StartSyncRunRequestBody requestBody,
         [FromHeader(Name = "X-Supported-Datawallet-Version")]
         ushort supportedDatawalletVersion, CancellationToken cancellationToken)
     {
+        var identity = await _identitiesRepository.FindByAddress(_userContext.GetAddress(), CancellationToken.None) ?? throw new NotFoundException(nameof(Identity));
+
+        identity.EnsureIdentityIsToBeDeleted();
```

Comment:
I don't get it. You removed a bunch of checks for the status of identities (see for example your changes in the RelationshipsController). What you want to do here is more or less what you removed in those other places. So why did you come up with a completely new way of solving this, instead of just copying the code from there?

Reply:
Implemented the same approach as in the `RelationshipsController`.